src/context/loaderAdmin.provider.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useContext, useMemo, useState } from "react";
+
+type LoadingContextType = {
+  loading: boolean;
+  showLoader: () => void;
+  hideLoader: () => void;
+};
+
+const LoadingContext = createContext<LoadingContextType>({
+  loading: false,
+  showLoader: () => {},
+  hideLoader: () => {}
+});
+
+const LoadingProvider = ({ children }: { children: React.ReactNode }) => {
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const showLoader = () => setLoading(true);
+  const hideLoader = () => setLoading(false);
+
+  const value = useMemo(
+    () => ({
+      loading,
+      showLoader,
+      hideLoader
+    }),
+    [loading]
+  );
```

Comment:
:art: I'm a little surprised the linter isn't complaining about this. It's correct, but usually the linter complains if there are any values missing from the dependency array. A more idiomatic way to structure this might be:
```
const showLoader = useCallback(() => setLoading(true));
const hideLoader = useCallback(() => setLoading(false));
const value = { loading, showLoader, hideLoader }
```
or 
```
const value = useMemo(
  () => ({
    loading,
    showLoader: () => setLoading(true),
    hideLoader: () => setLoading(false),
  }),
  [loading]
);
```